add(toggle-pane) : add `reset_state` to set other restore state

Jaehaks opened the pull request:
When restoring after max_preview, revert to the min-preview state
To resolve #94

Jaehaks committed:
add(toggle-pane) : add `reset_state` to set other restore state

When restoring after max_preview, revert to the min-preview state

sxyazi commented:
Thanks for the PR @Jaehaks!

I'd prefer not to add extra config. Is it possible to leverage the existing state to restore the view? When `toggle-pane` is called for the first time, its state should be saved to `st`.

Jaehaks commented:
This PR suggests returning to the reset state as the original behavior if we don't configure a default setup. Is that not enough?

sxyazi commented:
Yes, that means the user needs to configure it.

I think it would feel more natural if you enter mode A but switch to mode B
without exiting, and then when you exit mode B, it restores to mode A, that
should be the default behavior.

On Sat, Mar 22, 2025 at 5:46 PM Jaehaks ***@***.***> wrote:

> This PR suggests returning to the reset state as the original behavior if
> we don't configure a default setup. Is that not enough?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/yazi-rs/plugins/pull/95#issuecomment-2745183842>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEFWFIDFG5KP3E2OKWC3IED2VUWRHAVCNFSM6AAAAABZRPYUNKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBVGE4DGOBUGI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
> [image: Jaehaks]*Jaehaks* left a comment (yazi-rs/plugins#95)
> <https://github.com/yazi-rs/plugins/pull/95#issuecomment-2745183842>
>
> This PR suggests returning to the reset state as the original behavior if
> we don't configure a default setup. Is that not enough?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/yazi-rs/plugins/pull/95#issuecomment-2745183842>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEFWFIDFG5KP3E2OKWC3IED2VUWRHAVCNFSM6AAAAABZRPYUNKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONBVGE4DGOBUGI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

Jaehaks commented:
Ok i'll try it. please wait. i  am not good at programming

Jaehaks committed:
fix(toggle-pane) : restore to previous state after maximization

1. Problem

- Back in the day, executing maximization again would always reset it to
  the default state laid out in the [manager] section of `yazi.toml`
- 301f971 commit needs to additional configuration to change reset state

2. Improvement

- Every minimization operation, it would save that state and then revert
  back to it after maximizing

Jaehaks commented:
@sxyazi 
I commited second one, check please

Jaehaks committed:
fix(toggle-pane) : remove setup()

`bf_state` declared in entry() to remove setup()
